### DISCUSSION TITLE
how to set a header while using delegate option in virtualservice

### DISCUSSION THREAD

**Discussion by sanjaybaburaj** [istio/istio#47834] [Q&A]
is it possible to add a request header while delegating to a virtualservice

![image](https://github.com/istio/community/assets/32713188/0e7cefb4-58bc-4f37-bd24-c558e2a84038)

if so, could you please provide any sample on setting a request header to a delegare virtual service, I tried giving the request header, but its not showing any request header while checking the network console of browser,

I need to set a header - test: "canary" while dleegating it to the frontend virtualservice.

![image](https://github.com/istio/community/assets/32713188/9d5d0136-da6c-42a7-84d7-239d947a77e1)

**Comment by zirain** [comment]
please submit this in istio/istio repo.

**Reply by sanjaybaburaj** [reply]
could you please provide the url

**Reply by zirain** [reply]
https://github.com/istio/istio/discussions

**Reply by sanjaybaburaj** [reply]
@zirain  I had posted it in discussions only, I didn't understand ur reply, please help

**Reply by zirain** [reply]
<img width="291" alt="image" src="https://github.com/istio/istio/assets/4354057/6ffa56da-c0b4-40c3-b4b4-0aff6481516f">

please kindly do not submit discussion on istio/community repo.

**Reply by sanjaybaburaj** [reply]
@zirain,  ok, Kindly please help if its possible to set a request header when we delegate the request to a virtual service
